src/Microsoft.AspNet.Diagnostics/ErrorPageMiddleware.cs

```diff
@@ -182,31 +188,104 @@ private StackFrame StackFrame(Chunk line)
 
             int lineNumber = line.ToInt32();
 
-            return string.IsNullOrEmpty(file)
-                ? LoadFrame(string.IsNullOrEmpty(function) ? line.ToString() : function, string.Empty, 0)
-                : LoadFrame(function, file, lineNumber);
+            if (string.IsNullOrEmpty(file))
+            {
+                return GetStackFrame(
+                    // Handle stack trace lines like
+                    // "--- End of stack trace from previous location where exception from thrown ---"
+                    string.IsNullOrEmpty(function) ? line.ToString() : function,
+                    file: string.Empty,
+                    startLineNumber: 0,
+                    endLineNumber: 0);
+            }
+            else
+            {
+                return GetStackFrame(function, file, lineNumber, lineNumber);
+            }
         }
 
-        private StackFrame LoadFrame(string function, string file, int lineNumber)
+        // make it internal to enable unit testing
+        internal StackFrame GetStackFrame(string function, string file, int startLineNumber, int endLineNumber)
         {
-            var frame = new StackFrame { Function = function, File = file, Line = lineNumber };
+            var frame = new StackFrame { Function = function, File = file, Line = startLineNumber };
+
+            if (string.IsNullOrEmpty(file))
+            {
+                return frame;
+            }
+
+            IEnumerable<string> lines = null;
             if (File.Exists(file))
             {
-                var code = File.ReadLines(file);
-                ReadFrameContent(frame, code, lineNumber, lineNumber);
+                lines = File.ReadLines(file);
             }
+            else
+            {
+                // Handle relative paths and embedded files
+                var fileInfo = _options.FileProvider.GetFileInfo(file);
```

Comment:
`_fileProvider`. Add test to validate `null` `_options.FileProvider` results in a `PhysicalFileProvider` being used.